certora/specs/CER-15-Permit/CER-51-Permit-msg-sender.spec

```diff
@@ -72,18 +79,36 @@ hook CALL(uint g, address addr, uint value, uint argsOffset, uint argsLength, ui
     assert(executingContract != currentContract || addr != currentContract);
 }
 
-// This rule checks the property of interest "EVC can only be msg.sender during the self-call in the permit() function". Expected to fail on permit() function.
+// This rule checks the property of interest "EVC can only be msg.sender during the self-call in the permit() function. Expected to fail on permit() function.
+// To prove this for controlCollateral, we need the additionl assumption
+// that the EVC can never become an account controller. (See the rule after this one)
 rule onlyEVCCanCallCriticalMethod(method f, env e, calldataarg args)
   filtered {f -> 
     !isMustRevertFunction(f) &&
-    f.selector != sig:EthereumVaultConnectorHarness.permit(address,uint256,uint256,uint256,uint256,bytes,bytes).selector
+    f.selector != sig:EthereumVaultConnectorHarness.permit(address,uint256,uint256,uint256,uint256,bytes,bytes).selector &&
+    f.selector != sig:EthereumVaultConnectorHarness.controlCollateral(address, address, uint256, bytes).selector
   }{
     //Exclude EVC as being the initiator of the call.
     require(e.msg.sender != currentContract);
-
     f(e,args);
 
     assert(true);
 }
 
+// For onlyController we need the additional assumption that
```

Comment:
the EVC can never become controller (not collateral) for any address. the comment should point to CER-11

Reply:
The special case for `onlyController` is in `onlyEVCCanCallCriticalMethodOnlyController` and a comment above that rule (where we use this assumption) included a reference to CER-80 https://linear.app/euler-labs/issue/CER-80/collaterals-restrictions. Since this assumption is also mentioned in this comment, I added another reference to the issue just to make sure readers do not miss this!